### DISCUSSION TITLE
WhisperWriter: A small, simple dictation tool!

### DISCUSSION THREAD

**Discussion by savbell** [openai/whisper#1259] [Show and tell]
I kept running into issues trying to use the Windows Dictation tool, so I created my own version using Whisper: [WhisperWriter](https://github.com/savbell/whisper-writer)! ✏

In the configuration files, you can set a keyboard shortcut ("ctrl+alt+space" by default) that, when pressed, will start recording from your microphone until it detects a pause in your speech. The audio is then sent to the Whisper API for transcription and then automatically typed out into the active window. I also added some configuration for removing trailing periods or converting the text to all lowercase since that's the way I usually type.

Something simple like this works perfectly for me because my main speech-to-text use case is short phrases or sentences, such as a Google search or a quick reply to a text. I thought I'd share in case anyone else is fed up with the built-in Windows speech recognition software! 😉

**Comment by wdarea51** [comment]
Hello,

Can you describe how you got this to work.  I just downloaded everything required to run whisper but I can not figure out how to being to import audio files or get it to read from my microphone?

**Reply by savbell** [reply]
Hello! Can you please provide me with some more details about the steps you took to run the script? As well as what operating system you're using?

After you follow the instructions in the ["Installation"](https://github.com/savbell/whisper-writer#installation) part of the Readme, you can start the program by running `python run.py` in your terminal. Then, to start recording, you have to press the keyboard shortcut set in the [config file](https://github.com/savbell/whisper-writer/blob/main/src/config.json). It's `ctrl + alt + space` by default. That should open the recording window and begin taking in input from your microphone, stopping once there is a long enough pause in your speech. It will then send the recorded audio to Whisper to transcribe and then automatically type it out in the input field your cursor is active on.

I hope that helps! 😊

**Comment by themanyone** [comment]
I came up with a similar tool for Linux https://github.com/themanyone/voice_typing
I will link discussion here.

**Comment by PedroAVJ** [comment]
Oh my god, you're a godsend. Seriously!, I was looking for a tool like this and couldn't find anything other than the really bad Windows recognition tool that they built in, especially for someone like me with broken English, since it's not my native language. I can't believe someone hasn't made a SASS or an app that does this. I would pay good money for it.  There's a gap in the market here.
